### DISCUSSION TITLE
Is `argocd.argoproj.io/manifest-generate-paths` recursive?

### DISCUSSION THREAD

**Discussion by andrii-korotkov-verkada** [argoproj/argo-cd#18999] [Q&A] [answered]
Can this annotation have effect when files in subdirectories are changed?
For example, when kustomize is used, there would be multiple directories to watch besides where the application points at. I wonder if it's possible to just specify this annotation for parent directory of all of those, or if each directory needs to be listed individually?

**Comment by crenshaw-dev** [comment, marked as answer]
The list of "affected files" comes either from the SCM webhook payload or from `git diff --name-only`. In general, these are file paths relative to the repo root.

The glob patterns listed in the annotation are evaluated by [filepath.Match](https://pkg.go.dev/path/filepath#Match).

So if any pattern matches any affected file, the app will be refreshed.

So if you want to match everything in a directory recursively, you can use a value like this: `/some-dir/*`.

(Note this is based on a quick reading of the code. Probably needs a deep dive and some docs updates/examples.)

**Reply by andrii-korotkov-verkada** [reply]
```
'*'         matches any sequence of non-Separator characters
```
Hm, seems like it may not do a recursive match :(
Just tested that `filepath.Match("/*", "/abc/def")` returns false.

**Reply by milan-veljkovic** [reply]
Hi @andrii-korotkov-verkada ,

Does this mean that the manifest paths need to include every subdirectory in order to be part of the refresh paths? For example, if the application path is `/cluster/app` and `manifest-generate-paths: .`, changes in `/cluster/app/subdir/deployment.yaml` won't be picked up, right? Does every subdirectory need to be added explicitly to manifest-generate-paths?

**Reply by andrii-korotkov-verkada** [reply]
Yes. I recommend using rendered manifests into a single file or similar. Also, the new hydrator feature might do something similar.

**Reply by milan-veljkovic** [reply]
Thank you so much for clarifying! I found different answers to the same question in the argo-cd Slack channel, which made it a bit confusing.
